### PR TITLE
feat(docker): add support for pgid and puid env variables

### PR DESCRIPTION
**Pull request by Meierschlumpf**
<br/>
<div align="center">
  
  <h3>Homarr</h3>
</div>

**Thank you for your contribution. Please ensure that your pull request meets the following pull request:**

- [x] Builds without warnings or errors (``pnpm buid``, autofix with ``pnpm format:fix``)
- [x] Pull request targets ``dev`` branch
- [x] Commits follow the [conventional commits guideline](https://www.conventionalcommits.org/en/v1.0.0/)
- [x] No shorthand variable names are used (eg. ``x``, ``y``, ``i`` or any abbrevation)

### CODE REVIEW

**Comment by IIIdefconIII** [comment]
following this thread

File: scripts/entrypoint.sh [hunk 1]
```diff
@@ -7,4 +7,9 @@ mkdir -p /appdata/redis
 
 chown -R nextjs:nodejs /appdata
 
+DOCKER_SOCKET=/var/run/docker.sock
+if [ -f "$DOCKER_SOCKET" ]; then
+    chown -R nextjs:nodejs $DOCKER_SOCKET
+fi
+
```
Comment (manuel-rw) [inline scripts/entrypoint.sh:14]: Add a comment, why this chown is required

**Review by manuel-rw — CHANGES_REQUESTED** [review]

File: scripts/entrypoint.sh [hunk 2]
```diff
@@ -7,4 +7,9 @@ mkdir -p /appdata/redis
 
 chown -R nextjs:nodejs /appdata
 
+DOCKER_SOCKET=/var/run/docker.sock
```
Comment (Meierschlumpf) [inline scripts/entrypoint.sh:10]:
```suggestion
# Change owner to allow homarr to read docker socket when mounted
DOCKER_SOCKET=/var/run/docker.sock
```

**Comment by darkcloud784** [comment]
Won't chowning the socket also impact other things using said socket? IE portainer. If so, might want to think about setting a user field that someone can run the container as that will have access to said socket.

**Comment by IIIdefconIII** [comment]
> Won't chowning the socket also impact other things using said socket? IE portainer. If so, might want to think about setting a user field that someone can run the container as that will have access to said socket.

mma also think so we need to get PUID and GUID ad environment variables

**Comment by Meierschlumpf** [comment]
> Won't chowning the socket also impact other things using said socket? IE portainer. If so, might want to think about setting a user field that someone can run the container as that will have access to said socket.

You are right, I didn't know that. I guess then we'll need to add `PUID` and `PGID`
Reference for me: https://docs.linuxserver.io/general/understanding-puid-and-pgid/#why-use-these

**Comment by Meierschlumpf** [comment]
What do you think of that solution? We allow the user to change PUID and PGID to whatever they need and chown the volumes the same way and internally still use the `internal_user` to run the processes as we don't want to change the owner of all files in the `/app` directory when starting up the container. But that user is member of the `homarr` group which is owner of the volumes

File: scripts/entrypoint.sh [hunk 3]
```diff
@@ -1,10 +1,30 @@
...
 mkdir -p /appdata/redis
 
-chown -R nextjs:nodejs /appdata
+chown -R $PUID:$PGID /appdata
+# Allow homarr group to write to appdata folder
+chmod -R g+w /appdata
+
+# Change owner to allow homarr to read docker socket when mounted
+DOCKER_SOCKET=/var/run/docker.sock
+if [ -r "$DOCKER_SOCKET" ]; then
```
Comment (manuel-rw) [inline scripts/entrypoint.sh:24]: What happens if the socket is read-only? At least run a test

**Review by manuel-rw — CHANGES_REQUESTED** [review]

File: scripts/entrypoint.sh (hunk 3 above)
Comment (IIIdefconIII) [reply scripts/entrypoint.sh:24]: image
cant wait for it to fly in working, alread just changed the compose file

File: Dockerfile [hunk 4]
```diff
@@ -25,51 +25,47 @@ RUN corepack enable pnpm && pnpm build
 FROM base AS runner
 WORKDIR /app
 
-# gettext is required for envsubst
-RUN apk add --no-cache redis nginx bash gettext su-exec
+# gettext is required for envsubst, shadow for usermod, su-exec for running as non-root
+RUN apk add --no-cache redis nginx bash gettext su-exec shadow
```
Comment (Meierschlumpf) [inline Dockerfile:29]: `shadow` can be removed

File: Dockerfile [hunk 5]
```diff
@@ -25,51 +25,47 @@ RUN corepack enable pnpm && pnpm build
...
-RUN addgroup --system --gid 1001 nodejs
-RUN adduser --system --uid 1001 nextjs
-
 # Enable homarr cli
-COPY --from=builder --chown=nextjs:nodejs /app/packages/cli/cli.cjs /app/apps/cli/cli.cjs
+COPY --from=builder /app/packages/cli/cli.cjs /app/apps/cli/cli.cjs
 RUN echo $'#!/bin/bash\ncd /app/apps/cli && node ./cli.cjs "$@"' > /usr/bin/homarr
 RUN chmod +x /usr/bin/homarr
 
+ENV APPLICATION_PGID=7575
```
Comment (Meierschlumpf) [inline Dockerfile:40]: Can be removed

File: Dockerfile [hunk 6]
```diff
@@ -25,51 +25,47 @@ RUN corepack enable pnpm && pnpm build
...
+RUN addgroup --system --gid $APPLICATION_PGID homarr_app
```
Comment (Meierschlumpf) [inline Dockerfile:41]: Can be removed

**Review by manuel-rw — APPROVED** [review]

**Review by manuel-rw — APPROVED** [review]

**Pull request merged**